builtin/providers/circonus/resource_circonus_check_caql.go

```diff
@@ -0,0 +1,89 @@
+package circonus
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/circonus-labs/circonus-gometrics/api/config"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+const (
+	// circonus_check.caql.* resource attribute names
+	checkCAQLQueryAttr = "query"
+)
+
+var checkCAQLDescriptions = attrDescrs{
+	checkCAQLQueryAttr: "The query definition",
+}
+
+var schemaCheckCAQL = &schema.Schema{
+	Type:     schema.TypeSet,
+	Optional: true,
+	MaxItems: 1,
+	MinItems: 1,
+	Set:      hashCheckCAQL,
```

Comment:
Discussed: `hashCheckCAQL()` trims the whitespace of strings before hashing the value because the API provider is doing the same.  For attributes that frequently have HEREDOC input, this is an important usability requirement.